numpydoc/docscrape.py

```diff
@@ -236,10 +236,30 @@ def _parse_param_list(self, content):
 
         return params
 
+    _role = r":(?P<role>\w+):"
+    _funcbacktick = r"`(?P<name>(?:~\w+\.)?[a-zA-Z0-9_.-]+)`"
```

Comment:
It's confusing to mix the use of `\w` and explicit character classes:
```suggestion
    _funcbacktick = r"`(?P<name>(?:~\w+\.)?[\w.-]+)`"
```
Actually, I don't think the `~` or `.` should be obligatory.
```suggestion
    _funcbacktick = r"`(?P<name>(?:~?\w+\.)?[\w.-]+|\w+)`"
```

Reply:
The regexes appearing in these patterns were taken from the original regex.
```
    _name_rgx = re.compile(r"^\s*(:(?P<role>\w+):"
                           r"`(?P<name>(?:~\w+\.)?[a-zA-Z0-9_.-]+)`|"
                           r" (?P<name2>[a-zA-Z0-9_.-]+))\s*", re.X)
```

The patterns can be changed but I'd prefer that to be in a separate PR, since that will change the meaning.